### PR TITLE
Design options tests DNM for feedback

### PR DESCRIPTION
**Pull request by mjkkirschner**
@Pboyer,
apologies for the strange diff setup here, I wanted to save you the time looking at code you've already seen, so this diff only includes: 
11 tests 
4 dyn files
and changes I've made to log to the console while loading the serialized presets

does not include any of the Spike work for the states editor

### CODE REVIEW

File: test/DynamoCoreTests/PresetStatesTests.cs [hunk 1]
```diff
@@ -0,0 +1,301 @@
...
+            string openPath = Path.Combine(TestDirectory, @"core\PresetStates\NoStates.dyn");
+            ViewModel.OpenCommand.Execute(openPath);
+           
+            Assert.AreEqual(5, ViewModel.CurrentSpace.Nodes.Count);
+            Assert.AreEqual(ViewModel.HomeSpace.PresetsCollection.DesignStates.Count(), 0);
+
+        }
+
+          [Test]
+          [Category("Failure")]
```
Comment (mjkkirschner) [inline test/DynamoCoreTests/PresetStatesTests.cs]: this TODO and failure category is erroneous, this has been fixed, will remove

**Comment by riteshchandawar** [comment]
@holyjewsus It will be good if you add any TD to the PR which is related to Test Case addition so that we will be knowing which kind of test cases we are having in our system.

Thanks,
Ritesh

**Comment by mjkkirschner** [comment]
@riteshchandawar sorry about that! Will do from now on.

**Comment by pboyer** [comment]
@holyjewsus This is a good start.  The one thing to point out is you should use `DynamoModelTestBase` wherever possible as your base class.  It looks like you don't really need the `ViewModel` layer for any of this, so why include it?  

Note, also, that we will be removing all WPF dependencies from `DynamoCoreTests`, so this will be necessary to bring in your code.

**Pull request closed without merging**